instaclustr/resource_cluster.go

```diff
@@ -628,35 +628,17 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error{
 	d.SetId(cluster.ID)
 	d.Set("cluster_id", cluster.ID)
 	d.Set("cluster_name", cluster.ClusterName)
-	clusterProvider := make(map[string]interface{}, 0)
-	clusterProvider["name"] = cluster.DataCentres[0].Provider
-	d.Set("cluster_provider", clusterProvider)
 
-	baseBundle := make(map[string]interface{}, 0)
-	baseBundle["bundle"] = cluster.BundleType
+	clusterProvider := make(map[string]interface{}, 0)
+	mapstructure.Decode(cluster.Provider, &clusterProvider)
```

Comment:
**[Question]** Any particular reason to use fixed/hard decode over `WeakDecode()`? Surprised there was no issues with type mismatching.

Reply:
Tried that before from Arjun's suggestion but didn't work